platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -270,7 +270,7 @@ public FileCollection getAgentClasspath() {
     public Provider<String> getAsJvmArg() {
         return providers.provider(() -> {
             StringBuilder builder = new StringBuilder();
-            ArgumentAppender argument = new ArgumentAppender(builder, task.getWorkingDir());
+            ArgumentAppender argument = new ArgumentAppender(builder, task.getWorkingDir().getAsFile().get());
```

Comment:
:question: Shouldn't this be `task.getWorkingDir().map(...)`?

Reply:
We can also do it that way, let me try

Reply:
Done